docs/website/docs/plus/getting-started/advanced_tutorial.md

```diff
@@ -0,0 +1,305 @@
+
+```py
+# get a pipeline instance
+pipeline = current.entities().create_pipeline("my_pipeline")
+# get a destination instance
+destination = current.entities().create_destination("duckdb")
+```
+
+### Running pipelines with the predefined source with the runner
+
+`dlt+` provides a pipeline runner that is used when you run pipelines from the cli as described in the basic tutorial. The project context also provides a runner instance that you can use to run your pipelines in code. 
+
+```py
+# get the runner
+runner = current.runner()
+# run the pipeline "my_pipeline" from the currently active project
+runner.run_pipeline("my_pipeline")
+```
+
+### Accessing the catalog
+
+The catalog provides access to all explicitely defined datasets
+
+```py
+# get a dataset instance pointing to the default destination (first in dataset destinations list) and access data inside of it
+# for this to work this dataset must already exist physically
+dataset = current.catalog().dataset("my_pipeline_dataset")
+# get the row counts of all tables in the dataset as a dataframe
+print(dataset.row_counts().df())
+```
+
+:::tip
+Read the docs on how to access data in python in dlt datasets [here](../../../general-usage/dataset-access/dataset) to learn more about the available data access methods. You can browse and filter tables and get the data in various formats.
+:::
+
+### Writing data back to the catalog
+
+The datasets on the dlt+ catalogs may also be written to. The datasets provide `write` method that can be used to write data back to the catalog. You will be able to control which datasets may be written to with contracts in the future. Under the hood, `dlt+` will use an ad-hoc dlt pipeline to run this operation.
+
+:::warning
+Writing data back to the catalog is an experimental feature at this time and should be used with caution until it is fully stable.
+:::
+
+```py
+# get a dataset from the catalog (this dataset must already exist physically and be explicitely defined in the dlt.yml)
+dataset = current.catalog().dataset("my_pipeline_dataset")
+# write a dataframe to the dataset into the table "my_table"
+dataset.write(pd.DataFrame({"name": ["John", "Jane", "Jim"], "age": [30, 25, 35]}), table_name="my_table")
+```
+
+You can also iterate over dataframes or arrow tables in an existing table and write them to a new table in the same or another dataset:
+
+```py
+# get dataset from the catalog
+dataset = current.catalog().dataset("my_pipeline_dataset")
+
+# here we iterate over the existing items table and write each dataframe to the dataset into a new table
+def transform_frames():
+    # iterate of table `items` in the dataset in chunks of 1000 rows
+    for df in dataset.items.iter_df(chunk_size=1000):
+        # do something with the dataframe here
+        yield df
+
+# write the dataframe to the dataset into the table "my_new_table"
+dataset.write(transform_frames, table_name="my_new_table")
```

Comment:
Apparently I'm doing something wrong, because here `dataset.write` also gives me an error:

```
dataset.write(transform_frames, table_name="my_new_table")
TypeError: 'ReadableDBAPIRelation' object is not callable
```